pandas/core/base.py

```diff
@@ -990,13 +991,73 @@ def factorize(self, sort=False, na_sentinel=-1):
         from pandas.core.algorithms import factorize
         return factorize(self, sort=sort, na_sentinel=na_sentinel)
 
-    def searchsorted(self, key, side='left'):
-        """ np.ndarray searchsorted compat """
+    _shared_docs['searchsorted'] = (
+        """Find indices where elements should be inserted to maintain order.
 
+        Find the indices into a sorted %(klass)s `self` such that, if the
+        corresponding elements in `v` were inserted before the indices, the
+        order of `self` would be preserved.
+
+        Parameters
+        ----------
+        v : array_like
+            Values to insert into `a`.
+        side : {'left', 'right'}, optional
+            If 'left', the index of the first suitable location found is given.
+            If 'right', return the last such index.  If there is no suitable
+            index, return either 0 or N (where N is the length of `a`).
+
+        Returns
+        -------
+        indices : array of ints
+            Array of insertion points with the same shape as `v`.
+
+        See Also
+        --------
+        numpy.searchsorted
+
+        Notes
+        -----
+        Binary search is used to find the required insertion points.
+
+        Examples
+        --------
+        >>> x = pd.Series([1, 2, 3])
+        >>> x
+        0    1
+        1    2
+        2    3
+        dtype: int64
+        >>> x.searchsorted(4)
+        array([3])
+        >>> x.searchsorted([0, 4])
+        array([0, 3])
+        >>> x.searchsorted([1, 3], side='left')
+        array([0, 2])
+        >>> x.searchsorted([1, 3], side='right')
+        array([1, 3])
+        >>>
+        >>> x = pd.Categorical(['apple', 'bread', 'bread', 'cheese', 'milk' ])
+        [apple, bread, bread, cheese, milk]
+        Categories (4, object): [apple < bread < cheese < milk]
+        >>> x.searchsorted('bread')
+        array([1])     # Note: an array, not a scalar
+        >>> x.searchsorted(['bread'])
+        array([1])
+        >>> x.searchsorted(['bread', 'eggs'])
+        array([1, 4])
+        >>> x.searchsorted(['bread', 'eggs'], side='right')
+        array([3, 4])    # eggs before milk
+        """)
+
+    @Substitution(klass='IndexOpsMixin')
+    @Appender(_shared_docs['searchsorted'])
+    def searchsorted(self, v, side='left', *args):
         # FIXME in GH7447
-        # needs coercion on the key (DatetimeIndex does alreay)
+        # needs coercion on the key (DatetimeIndex does already)
         # needs tests/doc-string
-        return self.values.searchsorted(key, side=side)
+        com.validate_args(args, min_length=0, max_length=1)
```

Comment:
You're probably right that there are cases where we currently use `**kwargs` to swallow arguments, but this is something we should improve in new code :). e.g., for `.mean()`, we should probably be writing something like this:
```
def mean(self, axis, skipna=True, **unused_kwargs):
    if unused_kwargs.get('keepdims', False):  # False is the default value from NumPy
        raise ValueError(...)
```

From a user experience point of view, I think it's worse to swallow invalid arguments than it is to always raise unexpected errors. In the later case, users will quickly realize that `np.searchsorted` doesn't work on pandas objects. But if we silently accept invalid values, they'll think `np.searchsorted` works until much later they may discover their code was giving erroneous results. We try to avoid these types of bugs :).